src/server/terrain/structure_building_blocks.zig

```diff
@@ -0,0 +1,290 @@
+		childrenToResolve.append(.{.parentId = parentId, .colorName = colorName, .colorIndex = colorIndex, .childIndex = childIndex, .structureId = structureId});
+		return .{
+			.structure = undefined,
+			.chance = zon.get(f32, "chance", 1.0),
+		};
+	}
+};
+
+pub fn registerSBB(structures: *std.StringHashMap(ZonElement)) !void {
+	if(structureCache.capacity() != 0) {
+		std.log.err("Attempting to register new SBBs without resetting cache.", .{});
+		return error.AlreadyRegistered;
+	}
+	std.log.debug("Registering {} structure building blocks", .{structures.count()});
+	structureCache.ensureTotalCapacity(arenaAllocator.allocator, structures.count()) catch unreachable;
+	childrenToResolve = .init(main.stackAllocator);
+	{
+		var iterator = structures.iterator();
+		while(iterator.next()) |entry| {
+			const value = StructureBuildingBlock.initFromZon(entry.key_ptr.*, entry.value_ptr.*) catch continue;
+			const key = arenaAllocator.dupe(u8, entry.key_ptr.*);
+			const result = structureCache.getOrPut(arenaAllocator.allocator, key) catch unreachable;
+
+			if(result.found_existing) {
+				std.log.err("Ignoring duplicated structure building block: '{s}'", .{entry.key_ptr.*});
+				continue;
+			}
+			result.value_ptr.* = value;
+			std.log.debug("Registered structure building block: '{s}'", .{entry.key_ptr.*});
+		}
+	}
+	{
+		for(childrenToResolve.items) |entry| {
+			const parent = structureCache.getPtr(entry.parentId) orelse {
+				std.log.err("Could not find parent structure '{s}' for child resolution.", .{entry.structureId});
+				continue;
+			};
+			const child = structureCache.getPtr(entry.structureId) orelse {
+				std.log.err("Could not find child structure '{s}' for child resolution.", .{entry.structureId});
+				continue;
+			};
+
+			std.log.debug("Resolving child structure '{s}'->'{s}'->'{d}' to '{s}'", .{entry.parentId, entry.colorName, entry.childIndex, entry.structureId});
+			parent.children[entry.colorIndex].items[entry.childIndex].structure = child;
+			continue;
+		}
+		childrenToResolve.deinit();
+	}
+	std.log.debug("Registered {} structure building blocks", .{structureCache.count()});
+}
+
+pub fn registerChildBlock(numericId: u16, stringId: []const u8) void {
+	const index: u16 = @intCast(childBlockNumericIdMap.count());
+	childBlockNumericIdMap.put(arenaAllocator.allocator, numericId, index) catch unreachable;
+	// Take only color name from the ID.
+	var iterator = std.mem.splitBackwardsScalar(u8, stringId, '/');
+	const colorName = iterator.next().?;
+	childBlockStringId.append(arenaAllocator, arenaAllocator.dupe(u8, colorName));
+	std.log.debug("Child block '{s}' {} ('{s}' {}) ", .{colorName, index, stringId, numericId});
+}
+
+pub fn registerBlueprints(blueprints: *std.StringHashMap([]u8)) !void {
+	if(blueprintCache.capacity() != 0) {
+		std.log.err("Attempting to register new blueprints without resetting cache.", .{});
+		return error.AlreadyRegistered;
+	}
+
+	originBlockNumericId = parseBlock(originBlockStringId).typ;
```

Comment:
I generally don't like seeing function aliases, because it makes me think that it is a local function.
Especially here, where it's only used once it's more confusing than just writing the path in my opinion.